lib/influxdb/query/core.rb

```diff
@@ -11,8 +11,34 @@ def version
         ping.header['x-influxdb-version']
       end
 
+      def quote(param)
+        if param.is_a?(String)
+          "'" + param.gsub(/['"\\\x0]/, '\\\\\0') + "'"
+        elsif param.is_a?(Integer) || param.is_a?(Float) || param == true || param == false
+          param.to_s
+        else
+          throw "Unexpected parameter type #{p.class} (#{p.inspect})"
+        end
+      end
+
+      def query_builder(query, params)
+        if params.is_a?(Array)
+          # convert array to hash
+          params = params.each_with_object({}).with_index do |(param, hash), i|
+            hash[(i + 11).to_s] = param
+          end
+        else
+          params = params.each_with_object({}) do |(k, v), hash|
+            hash[k.to_s] = v # to_s because we may be passed symbols
+          end
+        end
+        query.gsub(/:([a-z0-9]+):/i) { |p| quote(params[p[1..-2]]) }
```

Comment:
Not a problem for me, but I figured someone might want to have `where time > :a:s` (where a is the time in seconds). If you pass `'12345s'` as a parameter, the quoter would turn that into `'12345s'` rather than `12345s`, which would trip up influxdb.